pkgs/top-level/python-packages.nix

```diff
@@ -15402,6 +15404,20 @@ self: super: with self; {
 
   tree-sitter = callPackage ../development/python-modules/tree-sitter { };
 
+  tree-sitter0_21 = callPackage ../development/python-modules/tree-sitter0_21 { };
```

Comment:
I know this is a bit late, but why are you copying and pasting the file instead of using overrideAttrs and just modifying the original package, that would make this a lot more maintainable and would ensure that the two versions nix expressions don't diverge from each other?

Reply:
> I know this is a bit late, but why are you copying and pasting the file instead of using overrideAttrs and just modifying the original package, that would make this a lot more maintainable and would ensure that the two versions nix expressions don't diverge from each other?

The argument to make the expressions different is exactly related to the potential divergence: I anticipate that in the future the two expressions will keep on diverging and diverging, especially since the developers of `tree-sitter` don't officially support this outdated `0.21` version.

Reply:
I guess you should at least make yourself the maintainer rather then fabaff. I also don't know how to feel about keeping the older version alive, I guess it's fine as long as their is still upstream support, but once upstream drops support we kinda wann drop it as well I think.

Reply:
> I guess you should at least make yourself the maintainer rather then fabaff. I also don't know how to feel about keeping the older version alive, I guess it's fine as long as their is still upstream support, but once upstream drops support we kinda wann drop it as well I think.

You are correct, this is not ideal, but this is related to the choice of `python3.pkgs.tree-sitter-languages` v.s `python3.pkgs.tree-sitter-make`. The later is not packaged in Nixpkgs yet, and [for a good reason](https://github.com/Freed-Wu/autotools-language-server/commit/f149843becfcfd6b2bb4a98eb1f3984c01d5fd33#r142659163) and the former has [this issue](https://github.com/grantjenks/py-tree-sitter-languages/issues/67) with the latest version of `python3.pkgs.tree-sitter`.

TBH only now I managed to explain it good enough :) The situation perhaps should have been explained in some comments.